pki/ejbca/api-docs/apiary.apib

```diff
@@ -0,0 +1,490 @@
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+        
++ Response 200 (application/json)
+
+        {
+          "status": {
+              "date": "2017-08-21T17:26:03.322000+00:00",
+              "reason": "CERTIFICATEHOLD"
+          }
+        }
+
++ Response 200 (application/json)
+
+        {
+          "status": {
+              "date": "1969-12-31T23:59:59.999000+00:00",
+              "reason": "NOTREVOKED"
+          }
+        }
+
+## Certificate [/ca/{cacn}/certificate/{certserial}]
+### Get the base64 pem certificate [GET]
+
++ Parameters
+  + cacn: CN=IOTmidCA,O=EJBCA IOT,C=SE (required, string): Certificate distinguished name
+  + certserial: 45A168F4EA978E7E (required, string): certificate serial number in hexadecimal
+
++ Response 400 (application/json)
+
+        {
+          "code": 400,
+          "message": "soap message: Could not find CA with name null and ID 1992101327",
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 404 (application/json)
+
+        {
+          "code": 404,
+          "message": "No certificate found",
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 200 (application/json)
+
+        {
+          "certificate": {
+              "certificate": null,
+              "keyStore": null,
+              "type": 0,
+              "certificateData": "MIICozCCAYugAwIBAgIIDbsv7z5yer..."
+          }
+        }
+
+### Revoke a certificate [DELETE /ca/{cacn}/certificate/{certserial}?{reason}]
+Removes a user certificate
+
++ Parameters
+  + cacn: CN=IOTmidCA,O=EJBCA IOT,C=SE (required, string): Certificate distinguished name
+  + certserial: 45A168F4EA978E7E (required, string): certificate serial number in hexadecimal
+  + reason: UNSPECIFIED (optional, enum) - one of REVOKATION_REASON enum values. default: UNSPECIFIED
+
+
++ Response 404 (application/json)
+
+        {
+          "code": 404,
+          "message": "Inexistent reason code",
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 400 (application/json)
+
+        {
+          "code": 400,
+          "message": "soap message: Could not find end entity certificate. Issuer CN=PROTOCOL, serialNo 4690394092950658023.",
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 400 (application/json)
+  
+        {
+          "code": 400,
+          "message": "soap message: Certificate is already revoked. Issuer: 'CN=PROTOCOL', serno: 5084496247959973.",
+          "moreinfo": "https://dojot.github.io/ejbca-rest/apiary_latest.html"
+        }
+
++ Response 200 (application/json)
+  
+        {
+          "status": 200,
+          "message": "ok"
+        }
+
+## CRL [/ca/{caname}/crl]
+
+### Create or update current CRL [PUT]
```

Comment:
I didn't understand the idea of this method. Is this an action? If so, the method should be something like POST /ca/{caname}/generate_crl